rpc/trace.go

```diff
@@ -0,0 +1,111 @@
+package rpc
+
+import (
+	"fmt"
+
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/jsonrpc"
+	"github.com/NethermindEth/juno/utils"
+	"github.com/NethermindEth/juno/vm"
+)
+
+type TransactionTrace struct {
+	ValidateInvocation    FunctionInvocation `json:"validate_invocation,omitempty"`
+	ExecuteInvocation     ExecuteInvocation  `json:"execute_invocation,omitempty"`
+	FeeTransferInvocation FunctionInvocation `json:"fee_transfer_invocation,omitempty"`
+
+	ConstructorInvocation FunctionInvocation `json:"constructor_invocation,omitempty"`
+	FunctionInvocation    FunctionInvocation `json:"function_invocation,omitempty"`
+}
+
+type ExecuteInvocation struct {
+	FunctionInvocation
+	RevertReason string `json:"revert_reason,omitempty"`
+}
+
+type FunctionInvocation struct {
+	FunctionCall
+	CallerAddress  *felt.Felt           `json:"caller_address,omitempty"`
+	ClassHash      *felt.Felt           `json:"class_hash,omitempty"`
+	EntryPointType EntryPointType       `json:"entry_point_type,omitempty"`
+	CallType       CallType             `json:"call_type,omitempty"`
+	Result         []felt.Felt          `json:"result,omitempty"`
+	Calls          []FunctionInvocation `json:"calls,omitempty"`
+	Events         []EventContent       `json:"events,omitempty"`
+	Messages       []MsgToL1            `json:"msg_to_l1,omitempty"`
+}
+
+// enum: EXTERNAL, L1_HANDLER, CONSTRUCTOR
```

Comment:
is this a directive to compiler or just a regular comment?

Reply:
Just a regular comment